### PR TITLE
BCK-6282 Improve the handling of None values from the Tcl interface for tohil

### PR DESCRIPTION
**Pull request by resuna**
Add an option "-nonevalue" to tohil::call to allow alternate none-value strings to be used.

### CODE REVIEW

**Comment by resuna** [comment]
> Looks good, but could you add a test that ensures `tohil::NONE` does not receive special treatment when a `-nonevalue` has been set?

Already in there:

test tohil_call-1.18 {call expecting arbitrary sentinel with tohil::NONE argument} \
	-body {
		set sentinel "gensym-[clock seconds]"
		tohil::exec {def a(x): return x == None}
		tohil::call -nonevalue $sentinel a tohil::NONE
	} \
	-result 0

**Review by lehenbauer — APPROVED** [review]

**Comment by lehenbauer** [comment]
> > Looks good, but could you add a test that ensures `tohil::NONE` does not receive special treatment when a `-nonevalue` has been set?
> 
> Already in there:

OK, cool, I saw that test but misread it.  Accepting PR.

File: Doc/reference/tohil_tcl_functions.rst [hunk 1]
```diff
@@ -22,10 +22,14 @@ from Tcl interpreter, the following commands are available:
...
    that will be passed to the function as named parameters.
 
+   If *-nonevalue word* is specified, then this overrides the default sentinel
+   string.
+
    When you use tohil::call, Tohil converts all of your arguments
-   to Python Unicode, unless an argument is comprised of the special
-   sentinel `tohil::NONE`, in which case the Python "None" data type
-   is substituted in place of that argument.
+   to Python Unicode, unless an argument is comprised of  special
```
Comment (bovine) [inline Doc/reference/tohil_tcl_functions.rst:29]: i think you're missing a word before "special"?

Comment (resuna) [reply Doc/reference/tohil_tcl_functions.rst:29]: "a"

Comment (resuna) [reply Doc/reference/tohil_tcl_functions.rst:29]: ALso reminds me I need to update the documentation for my other ticket.
